be/tool/beck.c

```diff
@@ -1256,15 +1281,20 @@ static void emap_act(struct action *act, struct m0_be_tx *tx)
 			       emap_val->er_start) >> adom->sad_babshift;
 		m0_ext_init(&ext);
 
+		m0_mutex_lock(&b.b_emaplock[id]);
 		rc = adom->sad_ballroom->ab_ops->
 			bo_reserve_extent(adom->sad_ballroom,
 					  tx, &ext,
 					  M0_BALLOC_NORMAL_ZONE);
 		if (rc != 0) {
+			m0_mutex_unlock(&b.b_emaplock[id]);
 			M0_LOG(M0_ERROR, "Failed to reserve extent rc=%d", rc);
 			return;
 		}
 
+		b.b_data += ((ext.e_end - ext.e_start) << adom->sad_babshift)
```

Comment:
For example, here, you are accessing and modifying member field of a global variable.
This is a multi-threaded program. Yes, this is under the "&b.b_emaplock[id])".
But this is still confusing.


And also when accessing the "&b.b_emaplock[id]".

Reply:
This is just for displaying data throughput.
Its okay if this value is approximate.